### ISSUE TITLE
typed attrs: onClick should take a function or throw erro

### ISSUE THREAD

**Issue by ncammarata** [flintjs/flint#71]
```js
<button onClick={alert()}> should throw an error
```

**Comment by ncammarata** [comment]
this brings up a bigger discussion on the pros and cons of being type-safe with everything we do. There are definitely benefits that are huge but it may be weird if you're not expecting it

**Comment by natew** [comment]
It would still be a Maybe (fn | false | undef) right? So we could throw errors for now on object/array/number.

**Comment by natew** [comment]
I guess it could be "falsy" though, falsy | fn
